__tests__/screens/ChannelListScreenV2/ChannelListScreenV2.test.js

```diff
@@ -68,3 +144,50 @@ describe('ChannelListScreenV2', () => {
     expect(mockRefreshAnonymousChannelList).toHaveBeenCalled();
   });
 });
+
+jest.mock('react-native-push-notification', () => ({
+  checkPermissions: jest.fn()
+  // Add other methods from PushNotification that you use
+}));
+
+describe('ChannelListScreenV2 useEffect', () => {
+  let alertSpy;
+
+  beforeEach(() => {
+    jest.clearAllMocks();
+    jest.useFakeTimers();
+    alertSpy = jest.spyOn(Alert, 'alert'); // Spy on Alert.alert
+  });
+
+  afterEach(() => {
+    alertSpy.mockRestore(); // Restore the original implementation after each test
+  });
+
+  it('should not show prompt if screen is not focused', async () => {
+    const mockIsFocused = jest.fn(() => false);
+    jest.mock('@react-navigation/native', () => ({
+      ...jest.requireActual('@react-navigation/native'),
+      useNavigation: () => ({
+        isFocused: mockIsFocused
+      })
+    }));
+
+    render(<ChannelListScreenV2 />, {wrapper: Store});
+    jest.advanceTimersByTime(20000); // Advance timers by 20 seconds
+    expect(PushNotification.checkPermissions).not.toHaveBeenCalled();
+  });
+
+  it('should handle authorizationStatus 1 with needsPermission', async () => {
+    PushNotification.checkPermissions.mockImplementation((callback) => {
+      callback({authorizationStatus: 1, alert: false, badge: true, sound: true});
+    });
+    StorageUtils.lastPromptNotification.get.mockReturnValue(jest.getRealSystemTime);
+
+    render(<ChannelListScreenV2 />, {wrapper: Store});
+    jest.advanceTimersByTime(20000);
+    expect(StorageUtils.lastPromptNotification.set).not.toHaveBeenCalled();
+    expect(Alert.alert).not.toBeCalled();
+  });
+
+  // Add similar tests for other cases and scenarios
+});
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

There seems to be a potential issue with the way you're mocking `@react-navigation/native` in your tests. You are re-mocking it inside individual test cases (line 168), which might not work as expected because Jest's mock calls are hoisted to the top of the code block. This means that they're executed before any other code, and re-mocking in this way could lead to inconsistent behavior.

Instead, consider using a variable that can be changed in each test case to simulate different return values for the `isFocused` function. Here's an example:

```diff
-    const mockIsFocused = jest.fn(() => false);
-    jest.mock('@react-navigation/native', () => ({
-      ...jest.requireActual('@react-navigation/native'),
-      useNavigation: () => ({
-        isFocused: mockIsFocused
-      })
-    }));

+    let isFocused = false;
+    jest.mock('@react-navigation/native', () => ({
+      ...jest.requireActual('@react-navigation/native'),
+      useNavigation: () => ({
+        isFocused: jest.fn(() => isFocused)
+      })
+    }));
+
+    // Then in your test case:
+    isFocused = false;
```

This way, you can control the return value of `isFocused` by changing the `isFocused` variable in each test case.


<!-- This is an auto-generated comment by OSS CodeRabbit -->